client/content-plugins/ComponentRegistry.js

```diff
@@ -37,23 +42,21 @@ export default class ComponentRegistry {
     const element = isExtension
       ? (await import(`extensions/content-${_type}s/${path}`)).default
       : (await import(`components/content-${_type}s/${path}`)).default;
-    const type = isQuestion(element.type)
-      ? processAnswerType(element.subtype)
-      : element.type;
-    const componentName = this._getName(type);
+    const id = element.templateId || getType(element);
+    const componentName = this._getName(id);
     _registry.push({ ...pick(element, _attrs), componentName, position });
     Vue.component(componentName, element.Edit);
-    if (element.Toolbar) Vue.component(getToolbarName(type), element.Toolbar);
+    if (element.Toolbar) Vue.component(getToolbarName(id), element.Toolbar);
   }
 
   all() {
     return cloneDeep(this._registry);
   }
 
-  get(type) {
-    if (!type) return this.all();
+  get(type, defaultToAll = true) {
+    if (!type) return defaultToAll ? this.all() : null;
```

Comment:
For what is this being used?

Reply:
In the [AddElement](https://github.com/ExtensionEngine/tailor/blob/15077f809605b0f25e002d24905e70d244b18629/client/components/common/tce-core/AddElement/index.vue#L106) component and in the system settings for listing all the elements. But, looking at it now, I should have just made a separate method that returns all of them.